tests/response/CustomSendResponse.integration.test.ts

```diff
@@ -78,13 +78,21 @@ function customSendResponse(responseParameters: ResponseParameters): void {
     if (statusCode) {
         response.statusCode = statusCode
     }
-    response.setHeader('Content-Type', 'application/json; charset=utf-8')
+    if (response.setHeader) {
+        response.setHeader('Content-Type', 'application/json; charset=utf-8')
+    } else if (response.header) {
+        response.header('Content-Type', 'application/json; charset=utf-8')
+    }
     const contextRecord = context as Record<string, unknown>
     if (contextRecord && contextRecord.customText) {
-        response.end(
-            Buffer.from(JSON.stringify(contextRecord.customText), 'utf8'),
-        )
-    } else {
-        response.end(Buffer.from(JSON.stringify(executionResult), 'utf8'))
+        if (response.end) {
+            response.end(JSON.stringify(contextRecord.customText))
+        } else if (response.send) {
+            response.send(JSON.stringify(contextRecord.customText))
+        }
+    } else if (response.end) {
+        response.end(JSON.stringify(executionResult))
+    } else if (response.send) {
+        response.send(JSON.stringify(executionResult))
     }
```

Comment:
```suggestion
    } else {
        if (response.end) {
            response.end(JSON.stringify(executionResult))
        } else if (response.send) {
           response.send(JSON.stringify(executionResult))
       }
   }
```
I think this represents the flow a bit better. You could also extract a function that tries to send a response via `end` or `send`.

Reply:
This is not advised by the "no-lonely-if" linting rule: https://eslint.org/docs/latest/rules/no-lonely-if